Makefile

```diff
@@ -55,11 +52,21 @@ help: ## Display this help.
 	@printf "$$(git rev-parse HEAD 2>/dev/null)" >$@
 	@test -n "$$(git status --porcelain --untracked-files=no)" && echo -dirty >>$@ || true
 
-agent-protocol-forwarder: .git-commit $(SOURCES)
-	$(GOOPTIONS) CGO_ENABLED=0 go build $(GOFLAGS) -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'" -o "$@" "cmd/$@/main.go"
+GOFLAGS += -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' \
+                     -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'"
 
-cloud-api-adaptor: .git-commit $(SOURCES)
-	$(GOOPTIONS) go build $(GOFLAGS) -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'" -o "$@" "cmd/$@/main.go"
+# Build tags required to build cloud-api-adaptor are derived from BUILTIN_CLOUD_PROVIDERS.
+# When libvirt is specified, CGO_ENABLED is set to 1.
+space := $() $()
+comma := ,
+cloud-api-adaptor: GOFLAGS += -tags $(subst $(space),$(comma),$(strip $(BUILTIN_CLOUD_PROVIDERS)))
+
+ifneq (,$(filter libvirt,$(BUILTIN_CLOUD_PROVIDERS)))
+cloud-api-adaptor: GOOPTIONS := $(subst CGO_ENABLED=0,CGO_ENABLED=1,$(GOOPTIONS))
+endif
+
+$(BINARIES): .git-commit $(SOURCES)
+	$(GOOPTIONS) go build $(GOFLAGS) -o "$@" "cmd/$@/main.go"
```

Comment:
When the cloud-api-adaptor target is called the GOFLAGS and GOOPTION are overridden only for the cloud-api-adatptor target? i.e. it won't take into account for the forwarder?

Reply:
Only `cloud-api-adaptor` have build constrains for cloud providers. `agent-protocol-forwarder` does not contains code that are related to cloud providers. So, we only need to specify build tags for `cloud-api-adaptor`.

Reply:
I notice that build tags are also used in another target such as `test-e2e`. I'll update Makefile.

https://github.com/confidential-containers/cloud-api-adaptor/blob/8b0a7a8de5cc458ae6b8255ef8ef476614cc7bca/Makefile#L77-L78